src/python/ensembl/io/genomio/assembly/status.py

```diff
@@ -0,0 +1,478 @@
+        else:
+            logging.warning(f"No assembly report found for accession(s) {accessions}. Exiting !")
+            sys.exit(0)
+
+    return combined_asm_reports
+
+
+def extract_assembly_metadata(assembly_reports: Dict[str, dict]) -> Dict[str, ReportStructure]:
+    """Function to parse assembly reports and extract specific key information on
+    status and related fields.
+
+    Args:
+        assembly_reports: Key value pair of core_name : assembly report.
+
+    Returns:
+        Parsed assembly report meta (core, meta).
+    """
+    parsed_meta = {}
+
+    for core, asm_report in assembly_reports.items():
+        asm_meta_info = ReportStructure()
+
+        # Mandatory meta key parsing:
+        asm_meta_info["Asm accession"] = asm_report["accession"]
+        asm_meta_info["Asm name"] = asm_report["assembly_info"]["assembly_name"]
+        asm_meta_info["Assembly type"] = asm_report["assembly_info"]["assembly_type"]
+        asm_meta_info["Asm status"] = asm_report["assembly_info"]["assembly_status"]
+        asm_meta_info["Species Name"] = asm_report["organism"]["organism_name"]
+        asm_meta_info["Taxon ID"] = asm_report["organism"]["tax_id"]
+
+        ## Non-mandatory meta key parsing:
+        assembly_meta_keys = asm_report["assembly_info"].keys()
+        organism_keys = asm_report["organism"].keys()
+
+        # check for genome_notes:
+        if "genome_notes" in assembly_meta_keys:
+            complete_notes = ", ".join(asm_report["assembly_info"]["genome_notes"])
+            asm_meta_info["Asm notes"] = complete_notes
+        else:
+            asm_meta_info["Asm notes"] = "NA"
+
+        # check for biosample:
+        if "biosample" in assembly_meta_keys:
+            asm_meta_info["Asm last updated"] = asm_report["assembly_info"]["biosample"]["last_updated"]
+        else:
+            asm_meta_info["Asm last updated"] = "NA"
+
+        # check for paired assembly:
+        if "paired_assembly" in assembly_meta_keys:
+            asm_meta_info["Paired assembly"] = asm_report["assembly_info"]["paired_assembly"]["accession"]
+        else:
+            asm_meta_info["Paired assembly"] = "NA"
+
+        # check for isolate/strain type:
+        if "infraspecific_names" in organism_keys:
+            organism_type_keys = asm_report["organism"]["infraspecific_names"].keys()
+            if "isolate" in organism_type_keys:
+                asm_meta_info["Isolate/Strain"] = asm_report["organism"]["infraspecific_names"]["isolate"]
+            elif "strain" in organism_type_keys:
+                asm_meta_info["Isolate/Strain"] = asm_report["organism"]["infraspecific_names"]["strain"]
+            else:
+                asm_meta_info["Isolate/Strain"] = "NA"
+        else:
+            asm_meta_info["Isolate/Strain"] = "NA"
+
+        parsed_meta[core] = asm_meta_info
+
+    return parsed_meta
+
+
+def generate_report_tsv(
+    parsed_asm_reports: dict,
```

Comment:
```suggestion
    parsed_asm_reports: Dict,
```
Same as before with `list`.